### PR TITLE
Mason doc, runtime/compile args, TOML update

### PR DESCRIPTION
**Pull request by Spartee**
This Pr includes three features: an update to the documentation on runtime and compile time args; the mason doc command, and an update to the TOML library.  Two of these features are listed and will by crossed off the mason issue #7106. 

First, an update to the documentation about how mason now supports forwarding all compile time and runtime arguments with examples that are hopefully clear. This was one of the first bits of feedback we got, so it was important that it got documented well.

Second, a first pass at a ``mason doc`` procedure has been introduced. It does not currently document dependencies, but does document the main file of the project. Documenting dependencies will be a "next step".

Third, the toml library has been adjusted so that if any Mason.lock file is generated, we now ensure that, for readability, the main project signified by "[root]" will always be printed at the top of the lock file. 

All local testing had passed for these changes.

### CODE REVIEW

File: tools/mason/mason.chpl [hunk 1]
```diff
@@ -116,10 +117,28 @@ proc masonClean() {
   runCommand('rm -rf target');
 }
 
+ 
+proc masonDoc(args) {
+  var toDoc = basename(getEnv('PWD'));
+  var project = toDoc + '.chpl';
+  if isDir('src/') {
+    if isFile('src/' + project) {
+      var command = 'chpldoc src/' + project;
```
Comment (ben-albrecht) [inline tools/mason/mason.chpl]: Note: `mason` will now depend on `chpldoc`, so this dependency should be marked in the top-level `Makefile`

File: doc/rst/tools/mason/mason.rst [hunk 2]
```diff
@@ -93,8 +93,38 @@ For example, after ``mason run [ options ]``, the project directory appears as s
 	  MyPackage
 	
    
+Projects can also be multiple files. Mason handles this by forwarding all arguments
```
Comment (ben-albrecht) [inline doc/rst/tools/mason/mason.rst]: Note that projects can be multiple files without getting compiler flags involved. `-M` is actually not necessary for the example below, but would be example for a case more like:

```
MyPackage/
  Mason.toml
  Mason.lock
  src/
  MyPackage.chpl
  utils/
    MyPackageUtils.chpl
```

Comment (ben-albrecht) [reply doc/rst/tools/mason/mason.rst]: I think it'd be good to document the handling of compiler flags / execution flags as a standalone topic. The usage of `-M` to use modules in subdirectories seems more FAQ-ish than appropriate for documentation. An FAQ at the bottom of the docs wouldn't be a bad idea.

Comment (Spartee) [reply doc/rst/tools/mason/mason.rst]: I think this is too important of a topic to push to the bottom of the documentation before we have compile flags in ``Mason.toml``.

**Pull request merged**